index.js

```diff
@@ -48,133 +53,180 @@ function togpx( geojson, options ) {
-            "@lat": c[1],
-            "@lon":c[0]
-          };
-          if (c[2] !== undefined) {
-            o.ele = c[2];
-          }
-          if (times && times[i]) {
-            o.time = times[i];
-          }
-          seg.trkpt.push(o);
-        });
-        o.trkseg.push(seg);
-      });
-      gpx.gpx.trk.push(o);
-      break;
-    // Polygons / Multipolygons
     case "Polygon":
     case "MultiPolygon":
-      o = {
-        "name": options.featureTitle(f.properties),
-        "desc": options.featureDescription(f.properties)
-      };
-      add_feature_link(o,f);
-      o.trkseg = [];
-      var coords = f.geometry.coordinates;
-      var times = f.properties ? f.properties.times : null;
-      if (f.geometry.type == "Polygon") coords = [coords];
-      coords.forEach(function(poly) {
-        poly.forEach(function(ring) {
-          var seg = {trkpt: []};
-          var i = 0;
-          ring.forEach(function(c) {
-            var o = {
-              "@lat": c[1],
-              "@lon":c[0]
+      // Geometry represented uniformly as MultiLineString
+      var coordsLists;
+      switch (f.geometry.type) {
+        case "LineString":   coordsLists = [f.geometry.coordinates]; break;
+        case "MultiPolygon": coordsLists = [].concat.apply([], f.geometry.coordinates); break;
+        default:             coordsLists = f.geometry.coordinates; break;
+      }
+      // Create gpx route
+      if (options["gpx.rte"]) { // include route
+        if (coordsLists.length === 1) {  // single route
+          var rte = mkRte(f, coordsLists[0]);
+          gpx.rte.push(rte);
+        } else { // multiple routes are handled individually using recursive call
+          coordsLists.forEach(function (coords) {
+            var pseudo_feature = {
+              "properties": f.properties,
+              "geometry": {type: "LineString", coordinates: coords}
             };
-            if (c[2] !== undefined) {
-              o.ele = c[2];
-            }
-            if (times && times[i]) {
-              o.time = times[i];
-            }
-            i++;
-            seg.trkpt.push(o);
+            var recurse_options = defaults({"gpx.trk": false}, options);
+            mapFeature(pseudo_feature, recurse_options, gpx);
           });
-          o.trkseg.push(seg);
-        });
-      });
-      gpx.gpx.trk.push(o);
+        }
+      }
+      // Create gpx track
+      if(options["gpx.trk"]) { // include track
+        var trk = mkTrk(f, coordsLists);
+        gpx.trk.push(trk);
+      }
       break;
     case "GeometryCollection":
       f.geometry.geometries.forEach(function (geometry) {
         var pseudo_feature = {
           "properties": f.properties,
           "geometry": geometry
         };
-        mapFeature(pseudo_feature);
+        mapFeature(pseudo_feature, options, gpx);
       });
       break;
     default:
       console.log("warning: unsupported geometry type: "+f.geometry.type);
     }
+  };
```

Comment:
the `;` is superfluous here, right? (as it's the end of the `mapFeature` function definition)